core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -128,12 +128,13 @@ private void init(long offset) throws BlockDoesNotExistException, IOException {
     UnderFileSystem ufs = mUfsManager.get(mBlockMeta.getMountId()).getUfs();
     ufs.connectFromWorker(
         NetworkAddressUtils.getConnectHost(NetworkAddressUtils.ServiceType.WORKER_RPC));
-    if (!ufs.isFile(mBlockMeta.getUnderFileSystemPath())) {
+    try {
+      updateUnderFileSystemInputStream(offset);
+    } catch (IOException e) {
       throw new BlockDoesNotExistException(
-          ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(mBlockMeta.getUnderFileSystemPath()));
+          ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(mBlockMeta.getUnderFileSystemPath()),
+          e);
```

Comment:
Is it guaranteed that the `IOException` will be because of the path not existing?

Reply:
it is not